src/player.cpp

```diff
@@ -3548,6 +3548,15 @@ void player::disp_status( WINDOW *w, WINDOW *w2 )
         veh = g->m.veh_at( pos() );
     }
     if( veh ) {
+        if(utf8_width(veh->name) < getmaxx(w)) {
+            mvwprintz( w, 1, getmaxx(w) - utf8_width(veh->name), c_ltgreen, veh->name.c_str());
+        } else {
+            // convert to a utf8_wrapper...
+            utf8_wrapper veh_name = veh->name.c_str();
+            // ...and print a shortened version to the screen
+            mvwprintz( w, 1, 0, c_ltgreen, veh_name.shorten(getmaxx(w)).c_str());
```

Comment:
For safety, this should be printed with `mvwprintz( ..., "%s", name.c_str() );`

If somebody names their loot mobile "The %**% thing", `mvwprintz` will interpret those `%` as format specifiers and will either fail (because they form an invalid format) or will try to access non-existing arguments ("The %%%s thing").

Same for the call above.

Also: the constructor of `utf8_wrapper` (two lines above) accepts a `std::string` as parameter, no need to call `.c_str()` when calling it.